python/pyiceberg/avro/writer.py

```diff
@@ -78,22 +79,25 @@ def write(self, encoder: BinaryEncoder, val: float) -> None:
 
 class DateWriter(Writer):
     def write(self, encoder: BinaryEncoder, val: Any) -> None:
-        encoder.write_date_int(val)
+        encoder.write_int(date_to_days(val))
```

Comment:
Here as well, the encoders and decoders should not use datetime objects.

Reply:
This is correct, right? The encoders only work with physical types. The `write_date_int` accepted a `date`, and now we do the conversion in the writer itself.

Reply:
It's fine to have a writer that work with date, but right now we need one that works with `int` because that's our internal representation. If someone passes a filter that is `d < DATE '2023-08-01'` for example, we'll receive that as something like this: `LessThan(Ref("d"), IntLiteral(1234))`. We want to compare integers rather than dates because that's faster and more reliable. When we read/write Avro, we want to go to that internal representation.